src/CSET/operators/regrid.py

```diff
@@ -179,3 +189,123 @@ def regrid_onto_xyspacing(
+    boundaries.]
+    NotImplementedError
+        If the cubes grid, or the method for regridding, is not yet supported.
+
+    Notes
+    -----
+    The acceptable coordinate names for X and Y coordinates are currently described
+    in X_COORD_NAMES and Y_COORD_NAMES. These cover commonly used coordinate types,
+    though a user can append new ones.
+    Currently rectilinear grids (uniform) are supported.
+
+    """
+    # Get a list of coordinate names for the cube
+    coord_names = [coord.name() for coord in incube.coords()]
+
+    # Check which x-coordinate we have, if any
+    x_coords = [coord for coord in coord_names if coord in X_COORD_NAMES]
+    if len(x_coords) != 1:
+        raise ValueError("Could not identify a unique x-coordinate in cube")
+    x_coord = incube.coord(x_coords[0])
+
+    # Check which y-coordinate we have, if any
+    y_coords = [coord for coord in coord_names if coord in Y_COORD_NAMES]
+    if len(y_coords) != 1:
+        raise ValueError("Could not identify a unique y-coordinate in cube")
+    y_coord = incube.coord(y_coords[0])
+
+    # List of supported grids - check if it is compatible
+    # NOTE: The "RotatedGeogCS" option below seems to be required for rotated grids --
+    #  this may need to be added in other places in these Operators.
+    supported_grids = (iris.coord_systems.GeogCS, iris.coord_systems.RotatedGeogCS)
+    if not isinstance(incube.coord(x_coord).coord_system, supported_grids):
+        raise NotImplementedError(
+            f"Does not currently support {incube.coord(x_coord).coord_system} regrid method"
+        )
+    if not isinstance(incube.coord(y_coord).coord_system, supported_grids):
+        raise NotImplementedError(
+            f"Does not currently support {incube.coord(y_coord).coord_system} regrid method"
+        )
+
+    # Get axis
+    lat, lon = incube.coord(y_coord), incube.coord(x_coord)
+
+    # Get bounds
+    lat_min, lon_min = lat.points.min(), lon.points.min()
+    lat_max, lon_max = lat.points.max(), lon.points.max()
+
+    # Get bounds
+    # Boundaries of frame to avoid selecting gridpoint close to domain edge
+    # Currently hardwired to 8 but could be a user input
+    lat_min_bound, lon_min_bound = lat.points[7], lon.points[7]
+    lat_max_bound, lon_max_bound = lat.points[-8], lon.points[-8]
+
+    # Check to see if selected point is outside the domain
+    if (
+        (lat_pt < lat_min)
+        or (lat_pt > lat_max)
+        or (lon_pt < lon_min)
+        or (lon_pt > lon_max)
+    ):
+        raise ValueError("Selected point is outside the domain.")
+    else:
+        if (
+            (lat_pt < lat_min_bound)
+            or (lat_pt > lat_max_bound)
+            or (lon_pt < lon_min_bound)
+            or (lon_pt > lon_max_bound)
+        ):
+            warnings.warn(
+                "Selected point is within 8 gridlengths of the domain edge.",
+                category=BoundaryWarning,
+                stacklevel=2,
+            )
```

Comment:
Switched to [warnings.warn](https://docs.python.org/3/library/warnings.html#warnings.warn) to allow testing of this code.

As we talked about previously, this is only logged, which may it may be missed if just looking at the workflow output. I've opened #583 to look at a better way to do this in future, but that will probably come after this has landed.